validphys2/src/validphys/config.py

```diff
@@ -693,6 +696,83 @@ def produce_experiment_from_input(
+    use_thcovmat_in_fitting=False,
+    use_t0_fitting=True,
+    ):
+        from validphys import covmats
+        if use_t0_fitting:
+            if theory_covmat_flag and use_thcovmat_in_fitting:
+                return covmats.dataset_inputs_t0_total_covmat
+            else:
+                return covmats.dataset_inputs_t0_exp_covmat
+        else:
+            if theory_covmat_flag and use_thcovmat_in_fitting:
+                return covmats.dataset_inputs_total_covmat
+            else:
+                return covmats.dataset_inputs_exp_covmat
+
+    @configparser.explicit_node
+    def produce_dataset_inputs_sampling_covmat(self,sep_mult,theory_covmat_flag=False,
+    use_thcovmat_in_sampling=False,
+    use_t0_sampling=False,
+    ):
+        from validphys import covmats
+        if use_t0_sampling:
+            if theory_covmat_flag and use_thcovmat_in_sampling:
+                if sep_mult:
+                    return covmats.dataset_inputs_t0_total_covmat_separate
+                else:
+                    return covmats.dataset_inputs_t0_total_covmat
+            else:
+                if sep_mult:
+                    return covmats.dataset_inputs_t0_exp_covmat_separate
+                else:
+                    return covmats.dataset_inputs_t0_exp_covmat
+        else:
+            if theory_covmat_flag and use_thcovmat_in_sampling:
+                if sep_mult:
+                    return covmats.dataset_inputs_total_covmat_separate
+                else:
+                    return covmats.dataset_inputs_total_covmat
+            else:
+                if sep_mult:
+                    return covmats.dataset_inputs_exp_covmat_separate
+                else:
+                    return covmats.dataset_inputs_exp_covmat
+
+    def produce_loaded_theory_covmat(self,output_path,
+    data_input,
+    theory_covmat_flag=False,
+    use_user_uncertainties=False,
+    use_scalevar_uncertainties=True
+    ):
+        if theory_covmat_flag is False:
+            return np.array([])
+        #Load correct file according to how the thcovmat was generated by vp-setupfit
+        generic_path = None
+        if use_user_uncertainties is True:
+            if use_scalevar_uncertainties is True:
+                generic_path = "datacuts_theory_theorycovmatconfig_total_theory_covmat.csv"
+            else:
+                generic_path = "datacuts_theory_theorycovmatconfig_user_covmat.csv"
+        else:
+            generic_path = "datacuts_theory_theorycovmatconfig_theory_covmat_custom.csv"
+
+        theorypath = output_path/"tables"/generic_path
+        theory_covmat = pd.read_csv(theorypath, index_col=[0, 1, 2], header=[0, 1, 2], sep="\t|,", engine="python").fillna(0)
+        #change ordering according to exp_covmat (so according to runcard order)
+        tmp = theory_covmat.droplevel(0, axis=0).droplevel(0, axis=1)
+        bb = [str(i) for i in data_input]
+        return tmp.reindex(index=bb, columns=bb, level=0).values
```

Comment:
```suggestion
        dataset_order = [str(i) for i in data_input]
        return theory_covmat.reindex(index=dataset_order, columns=dataset_order, level=1).values
```

This might fix it.